### PR TITLE
CLOUDP-277319: Added helm automation job #2

### PR DESCRIPTION
**Pull request by igor-karpukhin**
✅ [Test helm update](https://github.com/mongodb/mongodb-atlas-kubernetes/actions/runs/11502820493)

### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,7 +11,7 @@ DOCKER_SBOM_PLUGIN_VERSION=0.6.1
 # To re-generate a bundle for another specific version without changing the standard setup, you can:
 # - use the VERSION as arg of the bundle target (e.g make bundle VERSION=0.0.2)
 # - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
-VERSION ?= $(shell git describe --tags --dirty --broken | cut -c 2-)
+VERSION ?= $(shell git describe --always --tags --dirty --broken | cut -c 2-)
```
Comment (s-urbaniak) [inline Makefile:14]: what does `--always` change in terms of VERSION generation?

**Review by s-urbaniak — APPROVED** [review]

**Review by josvazg — COMMENTED** [review]
Does this finally work? I can't see a working run here:
https://github.com/mongodb/mongodb-atlas-kubernetes/actions/workflows/update-helm.yaml

**Comment by josvazg** [comment]
FYI [CLOUDP-278910](https://jira.mongodb.org/browse/CLOUDP-278910) CI GH App token action consistent usage is created for discuss

File: Makefile (hunk 1 above)
Comment (igor-karpukhin) [reply Makefile:14]: This makes sure that `git` always returns exit code 0. Without it, if will fail with "Nothing to describe" if there are no tags

**Comment by igor-karpukhin** [comment]
@josvazg unfortunately it doesn't. But as you can see, I found a way to generate token inside the Github Action, instead of encoding the key with base64 and then calling a makefile

**Comment by igor-karpukhin** [comment]
Reopened as there the workflow has been proved to be correct. The issue is with the credentials

**Comment by josvazg** [comment]
Does this actually work?

I fear there is no special permission to allow for a workflow in one repo to actually make a commit or PR on another repo. Instead, we will need to do this in 2 pieces:
- From the AKO repo we trigger a workflow in the helm-charts repo.
- The helm-charts repo workflow does the required update.
This 2 pieces flow is currently working with our Github App, when releasing AKO we trigger a creation of a PR in the helm charts repo.

If cross referencing repos has not been proved to work using a test Github App, I would assume it does not work. Using user level credentials might allow cross repo changes, but GitHub Apps might not get the same privileges, specially on org owned accounts.

**Comment by igor-karpukhin** [comment]
> If cross referencing repos has not been proved to work using a test Github App, I would assume it does not work
I can show you that it works in my two private repos

**Review by josvazg — APPROVED** [review]

**Comment by josvazg** [comment]
✅ [Test helm update](https://github.com/mongodb/mongodb-atlas-kubernetes/actions/runs/11502820493)
@s-urbaniak this is now working after updating permissions and commit e980c7a

**Pull request merged**